test/unit/test_url_util.py

```diff
@@ -0,0 +1,24 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+import pytest
+
+from snowflake.connector.url_util import is_valid_url, url_encode_str
```

Comment:
we also need to ignore the import error otherwise other driver just doesn't have this module.
```suggestion
try:
    from snowflake.connector.url_util import is_valid_url, url_encode_str
except ImportError:
    is_valid_url = None
    url_encode_str=None
```